pkg/manifests/manifests.go

```diff
@@ -2558,6 +2569,72 @@ func (f *Factory) NewValidatingWebhook(manifest io.Reader) (*admissionv1.Validat
 	return NewValidatingWebhook(manifest)
 }
 
+func (f *Factory) NewConsolePlugin(manifest io.Reader) (*consolev1.ConsolePlugin, error) {
+	return NewConsolePlugin(manifest)
+}
+
+func (f *Factory) MonitoringPlugin() (*consolev1.ConsolePlugin, error) {
+	return f.NewConsolePlugin(f.assets.MustNewAssetReader(MonitoringPlugin))
+}
+
+func (f *Factory) MonitoringPluginDeployment() (*appsv1.Deployment, error) {
+	d, err := f.NewDeployment(f.assets.MustNewAssetReader(MonitoringPluginDeployment))
+	if err != nil {
+		return nil, err
+	}
+
+	// ensure console-plugin container is present even if config isn't defined so that,
+	// we validate that deployment has the expected container name. Thereby avoiding
+	// any suprises should user add config later.
+	podSpec := d.Spec.Template.Spec
+	containers := podSpec.Containers
+	idx := slices.IndexFunc(containers, containerNameEquals(MonitoringPluginDeploymentContainer))
+	if idx < 0 {
+		return nil, errors.Errorf("failed to find console-plugin container: %s", MonitoringPluginDeploymentContainer)
```

Comment:
Maybe log where we failed to find the container too, i.e. the deployement?

Reply:
I think the convention we have in CMO is to either log or return error since errors are logged. @simonpasquier can confirm

Reply:
Sorry my mistake, I merely meant let's include where the container couldn't be found in the message. Didn't mean to request extra logging...apologies!

Reply:
yes I agree with @sthaha: you want to either return the error or log it but not both. And it's not to this repository but more a general Go idiom.

Reply:
The error would be something like 
```
UpdateConsolePluginFailed: failed to find console-plugin container: console-plugin
```
Doesn't that suffice?

Reply:
@jan--f, I will add the deployment info in the subsequent patch, since I have some modifications anyway